unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -16,108 +22,157 @@ interface UpdateReferralFeeProps {
           type="number"
           {...register('referralFeePercentage', {
             valueAsNumber: true,
-            min: 0,
+            min: 1,
             max: 100,
           })}
           error={
             errors?.referralFeePercentage &&
-            'This field accept percentage value between 0 and 100.'
+            'This field accept percentage value between 1 and 100.'
+          }
+          disabled={isDisabledReferrerInput}
+        />
+      </div>
+
+      <div className="grid gap-2">
+        <ToggleSwitch
+          title="Referrer address"
+          description={
+            'If no address is provided, the default referral fee will apply to any referrer.'
           }
-          disabled={disabledInput || !referralFee}
+          enabled={isReferralAddressToggled}
+          disabled={isDisabledReferrerInput}
+          setEnabled={(enabled: boolean) => {
+            setIsReferralAddressToggled(enabled)
+          }}
+        />
+
+        <AddressInput
+          label=""
+          value={referralAddress}
+          disabled={isDisabledReferrerInput || !isReferralAddressToggled}
+          onChange={(value: any) => {
+            setValue('referralAddress', value)
+          }}
+          error={errors?.referralAddress?.message}
+          onResolveName={onResolveName}
         />
       </div>
+
       {isManager && (
         <Button
           className="w-full md:w-1/3"
           type="submit"
-          disabled={disabledInput}
-          loading={setReferrerFeeMutation.isLoading}
+          disabled={
+            isDisabledReferrerInput ||
+            !referralFeePercentage ||
+            !!(referralAddress && !isValidAddress)
+          }
+          loading={isSettingReferrerFee}
         >
           Apply
         </Button>
       )}
+
+      <div className="flex flex-col mt-5 items-start">
+        <Card.Title>Referrers ({referralFees.length})</Card.Title>
+
+        {isLoading ? (
+          <div className="w-full flex items-center justify-center">
+            <LoadingIcon />
+          </div>
+        ) : referralFees.length ? (
+          <div className="w-full grid gap-5 grid-cols-1 mt-3">
+            {referralFees.map(({ id, referrer, fee }) => {
+              const feeNumber = Number(fee)
+              const isAddressLinkedToAnyReferrer = referrer === zeroAddress
+
+              return (
+                <div
+                  key={id}
+                  className="w-full overflow-x-auto flex justify-between items-center border border-gray-300 p-3 rounded-xl"
+                >
+                  <h2 className="text-md font-medium break-words">
+                    {isAddressLinkedToAnyReferrer
+                      ? 'Referral fee applied to any referrer'
+                      : referrer}
+                  </h2>
+                  <h2 className="text-md pl-3 font-medium break-words">
+                    ({feeNumber / 100}%)
+                  </h2>
+                </div>
```

Comment:
This one is saying basically is the address `0x0000000000000000000000000000000000000000`. This means it applies to any referrer.